inc/zoo/swar/SWAR.h

```diff
@@ -63,6 +66,7 @@ constexpr std::make_unsigned_t<T> lsbIndex(T v) noexcept {
 template<int NBits_, typename T = uint64_t>
 struct SWAR {
     using type = std::make_unsigned_t<T>;
+    using Boolean = BooleanSWAR<NBits_, T>;
```

Comment:
I don't like this forward reference.
It is not that intuition tells me that MSVC won't like this and give us subtle compilation problems because of MSVCs bugs, but something deeper.  I don't think that the abstraction of `BooleanSWAR` is in any way something that `SWAR` ought to advertise.  There's something fishy here that I can not express well.

Reply:
Yeah I wasn't sure about this either... wanted to see what you thought about this.